### PR TITLE
Policy based secret review system

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Description 📣

Simple review system to protect secrets from changing without project reviewers concern

REMINDER: This is unpolished version still missing

1. Close secret approval request feature
2. Stale ui
3. Conflicts ui
4. Proper messages on merge etc
5. Naming system for policy
6. Permission system

## Type ✨

- [x] New feature

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [ ] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝

### CODE REVIEW

**Comment by vmatsiiako** [comment]
I just tested approvals, and it looks/works really great! 🔥

**Comment by maidul98** [comment]
nit: would be nice to see the path where this secret action took place 
Screenshot 2023-10-04 at 9 40 37 PM

**Comment by maidul98** [comment]
nit: you don't know if you have approval requests. Would be nice to see a number
Screenshot 2023-10-04 at 9 43 54 PM

**Comment by maidul98** [comment]
Instead of showing old and new, i think we should show current and new, what do you think?

For example if there were requests 1 and 2 and we merge in request 2 first then request 1's old value is no longer the real old value because it has been changed by merging request # 2


Screenshot 2023-10-04 at 10 02 14 PM

**Comment by maidul98** [comment]
case: when a secret with the same name already exists and you request to create another secret with the same name, it looks like the new secret with the same name gets thrown away after being merged. 

I think we should be warning user that merging such change will mean overrideing the same key but with new value OR the request goes throw as a value update instead of as a secret create

**Comment by maidul98** [comment]
case: say there are two requests. request #1 and request #2. if request number one is a update and request number 2 is a delete and i merge in request number 2 first then i try to merge in request number 1, i get:
Screenshot 2023-10-04 at 10 12 23 PM

Screenshot 2023-10-04 at 10 12 29 PM


What we should say is that the updated secret has been deleted

File: backend/src/services/SecretApprovalService.ts
Comment (maidul98) [inline backend/src/services/SecretApprovalService.ts:365]: it looks like you are trying to account for when a secret already exists here (related to https://github.com/Infisical/infisical/pull/1044#issuecomment-1747930349)

File: backend/src/services/SecretApprovalService.ts
Comment (maidul98) [inline backend/src/services/SecretApprovalService.ts:601]: why is it important to save the conflicts and when do we make use of it?

**Comment by maidul98** [comment]
Also don't forget to remove the feature flag

File: backend/src/services/SecretApprovalService.ts
Comment (akhilmhdh) [reply backend/src/services/SecretApprovalService.ts:601]: This will tell the user what happend like why the create failed. Yah i was expecting to release this in next iteration. Ill just do it now then

File: frontend/public/lotties/system-outline-189-domain-verification.json
Comment (maidul98) [inline frontend/public/lotties/system-outline-189-domain-verification.json:1]: what is this file for?

**Review by maidul98 — APPROVED** [review]
amazing

**Comment by maidul98** [comment]
i don't think i saw any email notifications. We can add later though

**Pull request merged**
